### ISSUE TITLE
TextField Scale Animation Bug

### ISSUE THREAD

**Issue by estrode** [bp74/StageXL#205]
I updated to 12.0 this morning and started running into some issues with my application. After some searching, I isolated the bug to an issue when animating the scale of a sprite that contains a text field. When a sprite containing a text field animates its scale from a small value (0.1) or zero, the text field does not scale properly leaving the field distorted (if starting scale is small 0.1) or hidden (if starting scale is zero). Clicking on the text field after the animation is complete seems to fix the distortion and if the field can't be seen because the starting scale was 0, clicking will make the field appear. Hear is some sample code that demonstrates what I described above.

```dart
void main() {

  //Canvas is 600 x 600
  var canvas = html.querySelector('#stage');
  
  StageXL.stageOptions.backgroundColor = Color.Black;
  StageXL.stageOptions.renderEngine = RenderEngine.Canvas2D;
  var stage = new Stage(canvas);
  
  var renderLoop = new RenderLoop();
  var juggler = renderLoop.juggler;
  renderLoop.addStage(stage);
  
  var demoSprite = new Sprite();
  demoSprite.graphics.circle(0, 0, 100);
  demoSprite.graphics.fillColor(Color.White);
  
  var demoText = new TextField();
  demoText.defaultTextFormat = new TextFormat('Helvetica,Arial', 25, Color.Black);
  demoText.x = 0;
  demoText.y = 0;
  demoText.text = 'Test';
  demoSprite.addChild(demoText);
  
  //If scale is set to zero, the text field will be invisible until mouse click event
  demoSprite.scaleX = 0.1;
  demoSprite.scaleY = 0.1;
  
  stage.addChild(demoSprite);
  
  juggler.add(moveToAndScale(demoSprite, new Point(300, 300), 1));
}

Animatable moveToAndScale(Sprite sprite, Point point, num scale, [var seconds, var transitionFunc]) {
    if(seconds == null) seconds = 2;
    if(transitionFunc == null) transitionFunc = Transition.easeInCubic;

    AnimationGroup ag = new AnimationGroup();

    Tween moveTween = new Tween(sprite, seconds, transitionFunc);
    moveTween.animate.x.to(point.x);
    moveTween.animate.y.to(point.y);
    ag.add(moveTween);

    Tween scaleTween = new Tween(sprite, seconds, transitionFunc);
    scaleTween.animate.scaleX.to(scale);
    scaleTween.animate.scaleY.to(scale);
    ag.add(scaleTween);

    return ag;
}
```

**Comment by bp74** [comment]
Hi Eric,

That's an interesting issue. First of all - why does this happen: The problem is that StageXL 0.12 does automatically calculate the best size for the bitmap-cached TextField depending on the size of the TextField and the PixelRatio setting of the Stage (for HiDPI displays). Since the original scale is 0 the size of the TextField is 0 and therefore the cache has a size of 0.

A quick fix could be this:

    demoText.cacheAsBitmap = false;

This will disable the bitmap cache for the TextField and you will see a perfectly scaling TextField. The disadvantage is that rendering the TextField on every frame is much slower than the cached version.

It's a coincidence (you can call it a bug) that the TextField appears when you click on it. This is because TextField can also be used for input fields, and as soon as you click on it the cursor position is calculated and also the cache is refreshed - which will make the TextField visible.

I have to think a little bit how to fix this issue. Maybe we should automatically update the cache as soon as the size of the TextField chances?

**Comment by bp74** [comment]
Fixed in 6a7b48bb436b8fa0af3dabeac30a38cbf59d8f68
